src/transport_test.py

```diff
@@ -0,0 +1,179 @@
+"""Tests for transport."""
+
+import unittest
+from unittest.mock import MagicMock
+from unittest.mock import patch
+
+from gam import SetGlobalVariables
+import google_auth_httplib2
+import httplib2
+
+import transport
+
+
+class CreateHttpTest(unittest.TestCase):
+
+  def setUp(self):
+    SetGlobalVariables()
+    super(CreateHttpTest, self).setUp()
+
+  def test_create_http_sets_default_values_on_http(self):
+    http = transport.create_http()
+    self.assertIsNone(http.cache)
+    self.assertIsNone(http.timeout)
+    self.assertEqual(http.tls_minimum_version,
+                     transport.GC_Values[transport.GC_TLS_MIN_VERSION])
+    self.assertEqual(http.tls_maximum_version,
+                     transport.GC_Values[transport.GC_TLS_MAX_VERSION])
+    self.assertEqual(http.ca_certs, transport.GC_Values[transport.GC_CA_FILE])
+
+  def test_create_http_sets_tls_min_version(self):
```

Comment:
should we be testing with actual TLS version from:

https://docs.python.org/3/library/ssl.html#ssl.TLSVersion

here?

Reply:
We could, sure. I'm just validating the behavior that already existed.

What was the original purpose of saving these values as globals, rather than pulling them from the library directly, when used? Is GAM allowing users to set their own min/max through an environment var? I just want to avoid a case where the tests fail because the min/max values in the globals are different than that of the SSL lib.

Reply:
@jay0lee friendly ping. Any preference, here, on how this should behave in regards to the TLS lib defaults vs. the saved GAM globals?

Reply:
Google servers support TLS 1.2+ as should GitHub and the occassional other HTTPS server GAM talks to (few commands like set user photo can pull from an HTTP server). However there are two cases where admin may need something less than TLS 1.2:
- Some proxies need to force a certain older TLS version on the client (bad I know but it happens).
- some commands will pull from custom web server that may need older TLS.
GAM should default to requiring 1.2 or higher. It's possible one of above legit scenarios may require a downgrade and an environment variable can do that.

So all that to say GAM should probably throw an error when the value isn't one of the ones in `TLSVersion`. That change can come later but for now let's make sure the tests are using legit versions.

Reply:
OK. I think I misunderstood your original post on this. I get it now. Looks like the `ssl.TLSVersion` enum is only available in Python 3.7+, though, so using it here would cause some compatibility issues with 3.5 and 3.6 which I believe you're still testing.

`TLSVersion` is an [IntEnum](https://docs.python.org/3/library/enum.html#intenum), but I see you're [currently defaulting the min version to a string `'TLSv1_2'`](https://github.com/jay0lee/GAM/blob/master/src/var.py#L853). I'd be curious to know if that really works. Because, from the docs, it doesn't seem like it should. But, I have no other proof to back it up. For now, I could update the tests to use values like `'TLSv1_1'` and `'TLSv1_3'` to match what's there already.